### PR TITLE
Add Known* items to the local targeting pack resolution

### PR DESCRIPTION
**Pull request by ViktorHofer**
Add the required Known* items so that the ProcessFrameworkReferences
logic in the SDK knows about these and is able to create Resolved* items
that we later update to point to our local packs.

This is required when building with an SDK that doesn't support the
project's TargetFramework yet.

### CODE REVIEW

**Comment by ViktorHofer** [comment]
@Anipik this should hopefully unblock the issues that you were seeing in the libraries section. Ideally other parts of the repository that compile against net6.0 would use this logic as well but that required them to depend on the locally built targeting pack which might cause a dependency/sequencing issue today.

Let's see which projects in installer and mono actually require to be built against net6.0 already.

File: eng/targetingpacks.targets [hunk 1]
```diff
@@ -6,6 +6,33 @@
...
+                             RuntimeFrameworkName="$(SharedFrameworkName)"
+                             RuntimePackNamePatterns="$(SharedFrameworkName).Runtime.**RID**"
+                             RuntimePackRuntimeIdentifiers="linux-arm;linux-arm64;linux-musl-arm64;linux-musl-x64;linux-x64;osx-x64;rhel.6-x64;tizen.4.0.0-armel;tizen.5.0.0-armel;win-arm;win-arm64;win-x64;win-x86;ios-arm64;ios-arm;ios-x64;ios-x86;tvos-arm64;tvos-x64;android-arm64;android-arm;android-x64;android-x86;browser-wasm"
+                             TargetFramework="$(NetCoreAppCurrent)"
+                             TargetingPackName="$(SharedFrameworkName).Ref"
+                             TargetingPackVersion="$(ProductVersion)"
+                             Condition="'@(KnownFrameworkReference)' != '' and !@(KnownFrameworkReference->AnyHaveMetadataValue('TargetFramework', '$(NetCoreAppCurrent)'))" />
+    <KnownAppHostPack Include="$(SharedFrameworkName)" 
+                      AppHostPackNamePattern="$(SharedFrameworkName).Host.**RID**"
+                      AppHostPackVersion="$(MicrosoftNETCoreAppVersion)"
```
Comment (safern) [inline eng/targetingpacks.targets:24]: I guess this doesn't use ProductVersion as that one doesn't use the live built asset?

File: eng/targetingpacks.targets [hunk 2]
```diff
@@ -6,6 +6,33 @@
...
+                      AppHostRuntimeIdentifiers="linux-arm;linux-arm64;linux-musl-arm64;linux-musl-x64;linux-x64;osx-x64;rhel.6-x64;tizen.4.0.0-armel;tizen.5.0.0-armel;win-arm;win-arm64;win-x64;win-x86"
+                      TargetFramework="$(NetCoreAppCurrent)"
+                      Condition="'@(KnownAppHostPack)' != '' and !@(KnownAppHostPack->AnyHaveMetadataValue('TargetFramework', '$(NetCoreAppCurrent)'))" />
+    <KnownCrossgen2Pack Include="$(SharedFrameworkName).Crossgen2"
+                        TargetFramework="$(NetCoreAppCurrent)"
+                        Crossgen2PackNamePattern="$(SharedFrameworkName).Crossgen2.**RID**"
+                        Crossgen2PackVersion="$(ProductVersion)"
```
Comment (safern) [inline eng/targetingpacks.targets:31]: Does this one use the live pack?

File: eng/targetingpacks.targets (hunk 1 above)
Comment (ViktorHofer) [reply eng/targetingpacks.targets]: nope, that was just a mistake. Fixed. Thanks

File: eng/targetingpacks.targets (hunk 2 above)
Comment (ViktorHofer) [reply eng/targetingpacks.targets:31]: These are just defining the items that the ProcessFrameworkReferences target gets as an input. The routing to a locally built pack is done later in the custom UpdateTargetingAndRuntimePack target. That one currently doesn't know about `KnownAppHostPack` and `KnownCrossgen2Pack` as we don't need these yet.

**Comment by ViktorHofer** [comment]
Forgot to mention, but this will enable building a net[n].0 TargetFramework with a [n-1].0.100 SDK or to re-phrase, this will work in the future as well, not just for net6.0 :D

**Review by safern — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Hang is https://github.com/dotnet/runtime/issues/45894
Test failures are: https://github.com/dotnet/runtime/issues/45181

**Pull request merged**